gef-ui/server/controllers/amendments/facility-value/post-facility-value.test.ts

```diff
@@ -0,0 +1,220 @@
+
+import httpMocks from 'node-mocks-http';
+import { HttpStatusCode } from 'axios';
+import * as dtfsCommon from '@ukef/dtfs2-common';
+import { aPortalSessionUser, PORTAL_LOGIN_STATUS, DEAL_SUBMISSION_TYPE, DEAL_STATUS, Currency, PortalFacilityAmendmentWithUkefId } from '@ukef/dtfs2-common';
+import { postFacilityValue, PostFacilityValueRequest } from './post-facility-value';
+import { MOCK_BASIC_DEAL } from '../../../utils/mocks/mock-applications';
+import { MOCK_ISSUED_FACILITY } from '../../../utils/mocks/mock-facilities';
+import { FacilityValueViewModel } from '../../../types/view-models/amendments/facility-value-view-model';
+import { getCurrencySymbol } from './getCurrencySymbol';
+import { validationErrorHandler } from '../../../utils/helpers';
+import { validateFacilityValue } from './validation';
+import { getNextPage } from '../helpers/navigation.helper';
+import { PORTAL_AMENDMENT_PAGES } from '../../../constants/amendments';
+import { PortalFacilityAmendmentWithUkefIdMockBuilder } from '../../../../test-helpers/mock-amendment';
+
+jest.mock('../../../services/api', () => ({
+  getApplication: getApplicationMock,
+  getFacility: getFacilityMock,
+  updateAmendment: updateAmendmentMock,
+}));
+
+const dealId = 'dealId';
+const facilityId = 'facilityId';
+const amendmentId = 'amendmentId';
+
+const previousPage = 'previousPage';
+
+const mockUser = aPortalSessionUser();
+const userToken = 'userToken';
+
+const getHttpMocks = (facilityValue: string = '10000') =>
+  httpMocks.createMocks<PostFacilityValueRequest>({
+    params: {
+      dealId,
+      facilityId,
+      amendmentId,
+    },
+    session: {
+      user: mockUser,
+      userToken,
+      loginStatus: PORTAL_LOGIN_STATUS.VALID_2FA,
+    },
+    body: {
+      previousPage,
+      facilityValue,
+    },
+  });
+
+const mockDeal = { ...MOCK_BASIC_DEAL, submissionType: DEAL_SUBMISSION_TYPE.AIN, status: DEAL_STATUS.UKEF_ACKNOWLEDGED };
+
+describe('postBankReviewDate', () => {
+  let amendment: PortalFacilityAmendmentWithUkefId;
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.spyOn(dtfsCommon, 'isPortalFacilityAmendmentsFeatureFlagEnabled').mockReturnValue(true);
+
+    amendment = new PortalFacilityAmendmentWithUkefIdMockBuilder()
+      .withDealId(dealId)
+      .withFacilityId(facilityId)
+      .withAmendmentId(amendmentId)
+      .withChangeFacilityValue(true)
+      .build();
+
+    getApplicationMock.mockResolvedValue(mockDeal);
+    getFacilityMock.mockResolvedValue(MOCK_ISSUED_FACILITY);
+    updateAmendmentMock.mockResolvedValue(amendment);
+  });
+
+  it('should call getApplication with the correct dealId and userToken', async () => {
+    // Arrange
+    const { req, res } = getHttpMocks();
+
+    // Act
+    await postFacilityValue(req, res);
+
+    // Assert
+    expect(getApplicationMock).toHaveBeenCalledTimes(1);
+    expect(getApplicationMock).toHaveBeenCalledWith({ dealId, userToken: req.session.userToken });
```

Comment:
Please add console.error in the assertions where they should have been called and where they should have not been called.